libs/cli/langchain_cli/integration_template/integration_template/chat_models.py

```diff
@@ -14,50 +15,52 @@ class Chat__ModuleName__(BaseChatModel):
 
     Example:
         .. code-block:: python
+            
+            from langchain_core.messages import HumanMessage
 
             from __module_name__ import Chat__ModuleName__
 
-
             model = Chat__ModuleName__()
-    """
+            model.invoke([HumanMessage(content="Come up with 10 names for a song about parrots.")])
+    """  # noqa: E501
 
     @property
     def _llm_type(self) -> str:
         """Return type of chat model."""
         return "chat-__package_name_short__"
 
-    def _stream(
+    def _generate(
         self,
         messages: List[BaseMessage],
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
         **kwargs: Any,
+    ) -> ChatResult:
+        raise NotImplementedError
+
+    # TODO: Implement if __model_name__ supports streaming. Otherwise delete method.
+    def _stream(
+            self,
+            messages: List[BaseMessage],
+            stop: Optional[List[str]] = None,
+            run_manager: Optional[CallbackManagerForLLMRun] = None,
+            **kwargs: Any,
     ) -> Iterator[ChatGenerationChunk]:
         raise NotImplementedError
 
+    # TODO: Implement if __model_name__ supports async streaming. Otherwise delete
+    # method.
     async def _astream(
-        self,
-        messages: List[BaseMessage],
-        stop: Optional[List[str]] = None,
-        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
-        **kwargs: Any,
+            self,
+            messages: List[BaseMessage],
+            stop: Optional[List[str]] = None,
+            run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+            **kwargs: Any,
     ) -> AsyncIterator[ChatGenerationChunk]:
-        yield ChatGenerationChunk(
```

Comment:
I think removing these will cause mypy to fail. Also feels like a good template to help people mimic?

Reply:
Thoughts on template to mimic?

Reply:
but the actual implementation will only yield in for loop, this could confuse?

Reply:
Could switch to 
```
for chunk in ["Yield chunks", " like this!"]:
  yield ChatGenerationChunk(...)
```

Reply:
why have fake implementation for this method and not others